frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -124,6 +125,18 @@ def ensure_valid_state
       end
     end
 
+    def ensure_store_credits_eligible
+      expired_payments = @order.payments.valid.store_credits.select { |p| p.source.expired? }
```

Comment:
This amount of logic is awkward in a controller, especially since it's tied to the frontend. What about orders made through the admin, api, or a custom frontend?